torch/csrc/jit/script/sugared_value.cpp

```diff
@@ -363,8 +393,19 @@ RangeValue::RangeValue(
     throw ErrorReport(loc) << "range expected at most 3 arguments, got "
                            << inputs.size();
   }
+
+  // TODO: !has_only_end calculation
+  if (has_only_end_ && toIValue(end_)) {
```

Comment:
This should be implemented without the TODO or not at all. Not all all is perfectly fine (enumerate can still work by initializing the RangeValue directly with a constant). The intermediate states just bug reports, because we think it works when it really doesn't. This has happened with list comprehensions where since we didn't implement all of the spec the impression is that they do not work at all.